### PR TITLE
[5.5] Alternative return types in auth function phpdocs

### PR DESCRIPTION
**Pull request by aphofstede**
Tweaked phpdoc to "allow" alternative return types in overridden functions (e.g. SPAs wanting JSON instead of redirects)

### CODE REVIEW

**Comment by taylorotwell** [comment]
At this point we might as well just put "mixed" on it.

**Comment by aphofstede** [comment]
Ah yes, good ol' mixed! They have a common base-type, if you go all the way down to `\Symfony\Component\HttpFoundation\Response`, but I thought in most of these cases the Laravel versions would be more explicit. Thanks for reviewing.

File: src/Illuminate/Foundation/Auth/AuthenticatesUsers.php [hunk 1]
```diff
@@ -121,7 +121,9 @@ protected function authenticated(Request $request, $user)
      * Get the failed login response instance.
      *
      * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\RedirectResponse
+     * @return \Symfony\Component\HttpFoundation\Response
```
Comment (smhayhan) [inline src/Illuminate/Foundation/Auth/AuthenticatesUsers.php]: this function returns nothing, why did you return response class?

Comment (aphofstede) [reply src/Illuminate/Foundation/Auth/AuthenticatesUsers.php]: For overloading

**Pull request merged**